### ISSUE TITLE
Hookthrough Combo Tile Pointless (Entities)

### ISSUE THREAD

**Issue by Patiga** [ddnet/ddnet#2031]
The hookthrough combo tile (index 5 in the game layer)
https://github.com/ddnet/ddnet/blob/7d7b51011d5f64b634c808b416d9a24553569996/src/game/mapitems.h#L103
 is used to create hookthrough tiles. It does that by placing a unhookable block and overlays it with new hookthrough. 
However, it doesn't actually use the new hookthrough (index 66)
https://github.com/ddnet/ddnet/blob/7d7b51011d5f64b634c808b416d9a24553569996/src/game/mapitems.h#L142
, but instead places itself. 
In the game layer the tile looks like old hookthrough, while in the entity layer it looks just like new hookthrough. #2015 will make the two tiles behave exactly the same way.

Proposal:
1. Switch the tile visual representations of the combo tile (index 5) and the old hookthrough (index 6) in the game layer.
This would finally match the look of old hookthrough in the game layer and entities and the combo tile would have a sensible look in the game layer.
2. Remove the combo hookthrough from entities.
In the editor, actually place the new hookthrough on top of the unhookable block.
From the looks of it, it doesn't serve a new purpose, replace it with new hookthrough in existing maps.

game layer:
![game layer](https://raw.githubusercontent.com/ddnet/ddnet/7d7b51011d5f64b634c808b416d9a24553569996/data/editor/entities/DDNet.png)
entities:
![entities](https://raw.githubusercontent.com/ddnet/ddnet/7d7b51011d5f64b634c808b416d9a24553569996/data/editor/entities_clear/ddnet.png)

**Comment by 12pm** [comment]
the point of it is that you can replace both tiles at once

**Comment by heinrich5991** [comment]
@12pm 
> Remove the combo hookthrough from entities.
In the editor, actually place the new hookthrough on top of the unhookable block.
From the looks of it, it doesn't serve a new purpose, replace it with new hookthrough in existing maps.

**Comment by 12pm** [comment]
without using a special combo tile the editor cant know whether you intended to map a hookthrough or a hookblocker. if youd want to map it as a hookblocker and (accidentally) place it on an unhook, the editor would delete both tiles when you remove either of them. this is a rare case but it would introduce another bug and can be confusing and inconvenient.

as for the visual representation, i guess thats a different issue that the editor entities in general have.

**Comment by Ekidoxx** [comment]
Just please ping me in case any entities graphics will actually change, as in my spare time I'm working on updated version...

**Comment by Patiga** [comment]
@12pm  well instead of bug, lets call it an editor rule :)

Alternatively, how about removing both tiles when you remove the unhookable, while keeping the unhookable when removing the hookthrough.
Phrased differently: _**_Remove entire blocks (hookable, unhookable, hookthrough-combo) in the game layer, toggle secondary stuff like hookthrough in the front layer._**_
**Why?**
\+ If you accidentally place hookthrough on a unhookable, just remove it. 
\+ If you want to toggle hookthrough on a unhookable, do so in the front layer.
\+ Want to remove a hookthrough block? Remove it in the game layer like hookable/unhookable blocks.
\- Accidentally placed a unhookable onto a hookthrough? Well in this one case, bad luck. You just created a hookthrough block. Remove it in the game layer, replace your hookblock/hookthrough. Keep in mind that this is not the primary purpose of hookthrough.

Furthermore, this way mapping new hookthrough tiles would be consistent, not matter which map of which mapper you are editing.

**Comment by Patiga** [comment]
According to my [script](https://gitlab.com/Patiga/twmap/-/blob/master/src/bin/check_hookthrough.rs), there are 2 maps in total with misplaced combo-hookthrough.
> "ddmax/maps/Black_City.map": (208, 285)
"ddmax/maps/Black_City.map": (209, 285)
"ddmax/maps/Black_City.map": (210, 285)
"ddmax/maps/Hard Way 1.0.map": (143, 217)

It appears that those 4 tiles were just placed by accident, at a point in time at which the new hookthrough didn't exist yet. You can find the faulty tiles by loading the map and drawing a square with those coordinates starting from the upper left corner of the map.

**Comment by fokkonaut** [comment]
You need the combo tile, otherwise the client cant predict the solid block.

**Comment by heinrich5991** [comment]
Please read the proposal.

**Comment by BannZay** [comment]
Wait what? Hookstopper behaves as HookThrough when you place it on block? As a mapper I would say this is nonsense. Just let hookstopper be a hookstopper but not a frankenstein

**Comment by BannZay** [comment]
P.s.: I dont care how they looks like, similar or not. Graphics is a thing that could be changed at any time but behaviour is important thing. And this important things just must have strong reason to be so weird. Does anybody knows them?

**Comment by BannZay** [comment]
And yea, graphics can not be changed if you fucked up with entities behavior. Since we have frankenstein users can no longer make different graphics for opposite things

**Comment by Patiga** [comment]
I agree that having the 'new hookthrough' just be hookblock and the combo tile being the 'hookthrough block' would be a clean solution. Honestly don't know how much confusion this might cause.
I extended my [program](https://gitlab.com/Patiga/twmap/-/blob/master/src/bin/check_hookthrough.rs) to also count hookthrough instances which use the combo tile (id 5) and those that use the new hookthrough tile (id 66):
```
...
62577 combo tiles.
6433 new hookthrough tiles (excluding hookblock).
```
Those are quite many, would need to be automated. My tool isn't quite there yet, it is read-only so far  :)

**Comment by Patiga** [comment]
I don't think resaving so many maps is worth it. The tile is established now, and changing stuff will be more work than 'cleaning' anything up.
